### ISSUE TITLE
Could you consider using GitHub for version control and changes tracking?

### ISSUE THREAD

**Issue by eimix** [REPALphilread/ReP_AL-3D-Lawn-Mower#9]
Could you consider using GitHub for version control and changes tracking?

Now you use your own versionning 6.31 6.4 and so on, in this way could not see what changes were made.

Do you accept contribution to this project?

**Comment by baststar** [comment]
Hi Phil,
i refactored your structure to work within platformio (same codebase as of 6.7)
would be nice if you can use this structure, working with platformio/vscode is much easier, easy lib-updates since they are included via platformio.ini etc...

have a look here:
https://github.com/baststar/mowercode

had to put all NodeMCU-Code into one file because of the Blynk-Library which instantiates "Blynk" within their library :/

**Comment by eimix** [comment]
Looks like most builders are happy with Arduino.
Is it possible to use same "fork" for Platformio and Arduino?

**Comment by baststar** [comment]
For PlatformIO its required that you use cpp instead of .ino-Files. The codebases are not compatible. Ino-files are concated to one big file when the project is built in the ArduinoIDE (you need to import dependencies only once in the main file for example)

**Comment by eimix** [comment]
There is no coding comunity for this project - just builders :) 
it would be great to have some fork/branch with many code contributors, 
but  if using Platformio - not all users could use it straight forward.

All Platformio users understand Arduino but not the other way :/

**Comment by baststar** [comment]
yeah i understand, but its not so extremly different, just the structure. Its easier for new contributors with the platformio-structure, because you only need to clone the repository and can than just build it, without struggeling with libraries and other stuff. They are defined in the platformio.ini-file per project (mega-project, nano-project, uno-project and nodemcu-project). With the config.h-files you can merge updates without overwriting your configuration etc.
The code itself - the syntax - is the same, i just outsourced some declarations and initialisations etc
